### PR TITLE
[CPDLP-1025] Add NPQ statement selector

### PR DESCRIPTION
**Pull request by asmega**
## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CPDLP-1025

- Add NPQ statement selector component
- Pre-selects values to whatever the current statement is
- LHS Nav is gone and now full width

## Screenhots

Screenshot 2022-02-24 at 14 58 24

### Code quality checks
- [x] All commit messages are meaningful and true
- [x] Added enough automated tests

### HTML Checks
- [ ] All new pages have automated accessibility checks
- [ ] All new pages have visual tests via Percy

### Gotchas
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?

- Visit https://ecf-review-pr-1751.london.cloudapps.digital/
- Login as a finance user
- View an NPQ statement
- Should be able to switch to a different provider and/or statement without leaving the page

## External API changes

- None

### CODE REVIEW

File: app/models/finance/statement.rb [hunk 1]
```diff
@@ -20,6 +21,11 @@ def past_deadline_date?
   def current?
     payment_date > Time.current && deadline_date > Time.current
   end
+
+  def identifier
```
Comment (StupidCodeFactory) [inline app/models/finance/statement.rb:25]: Could simplify this all bit with `#to_param` https://api.rubyonrails.org/classes/ActiveRecord/Integration.html#method-i-to_param:

```suggestion
  def to_param
```

File: app/views/finance/npq/course_payment_breakdowns/show.html.erb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <% content_for :title, t("finance.npq.payment_breakdown") %>
-<% content_for :before_content, govuk_back_link(text: "Back", href: finance_npq_lead_provider_statement_path(@npq_lead_provider, @statement)) %>
+<% content_for :before_content, govuk_back_link(text: "Back", href: finance_npq_lead_provider_statement_path(@npq_lead_provider, @statement.identifier)) %>
```
Comment (StupidCodeFactory) [inline app/views/finance/npq/course_payment_breakdowns/show.html.erb:2]: can ✂️  with `to_param` as described  above

**Review by StupidCodeFactory — APPROVED** [review]

File: app/models/finance/statement.rb (hunk 1 above)
Comment (asmega) [reply app/models/finance/statement.rb]: done

File: app/views/finance/npq/statements/voided.html.erb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <% content_for :title, t("finance.voided_declarations") %>
-<% content_for :before_content, govuk_back_link(text: "Back", href: finance_npq_lead_provider_statement_path(@npq_lead_provider.id, id: @statement.id)) %>
+<% content_for :before_content, govuk_back_link(text: "Back", href: finance_npq_lead_provider_statement_path(@npq_lead_provider.id, id: @statement.identifier)) %>
```
[the remark this answers is not in the document]
Comment (asmega) [reply app/views/finance/npq/statements/voided.html.erb]: done

**Pull request merged**